### ISSUE TITLE
"Some cop" misses the `&block` / `block` variable

### ISSUE THREAD

**Issue by stdedos** [rubocop/rubocop#13368]
```ruby
class Array
  def map_in_parallel(&block)
    return self.map(&block) if defined?($disable_parallel_array_map) && $disable_parallel_array_map

    self.map do |item|
      block.call(item)
    end
  end
end

result = [1, 2, 3].map_in_parallel { |x| x * 2 }
puts result.inspect  # Output should be [2, 4, 6]
```
## Expected behavior

Describe here how you expected RuboCop to behave in this particular situation.

## Actual behavior

Describe here what actually happened.
Please use `rubocop --debug` when pasting rubocop output as it contains additional information.

```console
$ ./rubocop -a t.rb 
/home/stdedos/Documents/RelexOy/planning-cloud/backend/bin/bundle:7: warning: unused literal ignored
t.rb:1:1: C: Style/Documentation: Missing top-level documentation comment for class Array.
class Array
^^^^^^^^^^^
t.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
class Array
^
t.rb:2:23: C: [Corrected] Naming/BlockForwarding: Use anonymous block forwarding.
  def map_in_parallel(&block)
                      ^^^^^^
t.rb:2:23: C: [Corrected] Style/ArgumentsForwarding: Use anonymous block arguments forwarding (&).
  def map_in_parallel(&block)
                      ^^^^^^
t.rb:3:12: C: [Corrected] Style/RedundantSelf: Redundant self detected.
    return self.map(&block) if defined?($disable_parallel_array_map) && $disable_parallel_array_map
           ^^^^
t.rb:3:16: C: [Corrected] Naming/BlockForwarding: Use anonymous block forwarding.
    return map(&block) if defined?($disable_parallel_array_map) && $disable_parallel_array_map
               ^^^^^^
t.rb:3:16: C: [Corrected] Style/ArgumentsForwarding: Use anonymous block arguments forwarding (&).
    return map(&block) if defined?($disable_parallel_array_map) && $disable_parallel_array_map
               ^^^^^^
t.rb:3:31: C: Style/GlobalVars: Do not introduce global variables.
    return map(&) if defined?($disable_parallel_array_map) && $disable_parallel_array_map
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
t.rb:3:63: C: Style/GlobalVars: Do not introduce global variables.
    return map(&) if defined?($disable_parallel_array_map) && $disable_parallel_array_map
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
t.rb:5:5: C: [Corrected] Style/ExplicitBlockArgument: Consider using explicit block argument in the surrounding method's signature over yield.
    map do |item| ...
    ^^^^^^^^^^^^^
t.rb:5:5: C: [Corrected] Style/RedundantSelf: Redundant self detected.
    self.map do |item|
    ^^^^
t.rb:6:7: C: [Corrected] Performance/RedundantBlockCall: Use yield instead of block.call.
      block.call(item)
      ^^^^^^^^^^^^^^^^
t.rb:12:20: C: [Corrected] Layout/ExtraSpacing: Unnecessary spacing detected.
puts result.inspect  # Output should be [2, 4, 6]
                   ^
 1/1 file |======================= 100 =======================>| Time: 00:00:01 

1 file inspected, 13 offenses detected, 9 offenses corrected, 1 more offense can be corrected with `rubocop -A`
$ cat t.rb 
class Array
  def map_in_parallel(&)
    return map(&) if defined?($disable_parallel_array_map) && $disable_parallel_array_map

    map(&block)
  end
end

result = [1, 2, 3].map_in_parallel { |x| x * 2 }
puts result.inspect # Output should be [2, 4, 6]
```

## Steps to reproduce the problem

This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ [bundle exec] rubocop -V
1.67.0 (using Parser 3.3.5.0, rubocop-ast 1.32.3, analyzing as Ruby 3.3, running on ruby 3.3.5) [x86_64-linux]
  - rubocop-performance 1.22.1
  - rubocop-rspec 3.1.0
```
